lib/streamlit/runtime/caching/hashing.py

```diff
@@ -451,6 +451,46 @@ def _to_bytes(self, obj: Any) -> bytes:
                 # it contains unhashable objects.
                 return b"%s" % pickle.dumps(obj, pickle.HIGHEST_PROTOCOL)
 
+        elif type_util.is_type(obj, "polars.series.series.Series"):
+            import polars as pl  # type: ignore[import-not-found]
+
+            obj = cast(pl.Series, obj)
```

Comment:
nit: you could also include the length of the dimension of the series via `obj.shape` or `obj.len()`

Reply:
`shape` added!